src/Microsoft.SqlTools.ServiceLayer/LanguageExtensibility/ExternalLanguageService.cs

```diff
@@ -0,0 +1,233 @@
+    {
+        private ExternalLanguageOperations serviceOperations = new ExternalLanguageOperations();
+        private ConnectionService connectionService = null;
+        private static readonly Lazy<ExternalLanguageService> instance = new Lazy<ExternalLanguageService>(() => new ExternalLanguageService());
+
+        /// <summary>
+        /// Gets the singleton instance object
+        /// </summary>
+        public static ExternalLanguageService Instance
+        {
+            get { return instance.Value; }
+        }
+
+        /// <summary>
+        /// Internal for testing purposes only
+        /// </summary>
+        internal ConnectionService ConnectionServiceInstance
+        {
+            get
+            {
+                if (connectionService == null)
+                {
+                    connectionService = ConnectionService.Instance;
+                }
+                return connectionService;
+            }
+
+            set
+            {
+                connectionService = value;
+            }
+        }
+
+        public ExternalLanguageOperations ExternalLanguageOperations
+        {
+            get
+            {
+                return serviceOperations;
+            }
+            set
+            {
+                serviceOperations = value;
+            }
+        }
+
+        public void InitializeService(ServiceHost serviceHost)
+        {
+            serviceHost.SetRequestHandler(ExternalLanguageStatusRequest.Type, this.HandleExternalLanguageStatusRequest);
+            serviceHost.SetRequestHandler(ExternalLanguageListRequest.Type, this.HandleExternalLanguageListRequest);
+            serviceHost.SetRequestHandler(ExternalLanguageDeleteRequest.Type, this.HandleExternalLanguageDeleteRequest);
+            serviceHost.SetRequestHandler(ExternalLanguageUpdateRequest.Type, this.HandleExternalLanguageUpdateRequest);
+        }
+
+        /// <summary>
+        /// Handles external language delete request
+        /// </summary>
+        /// <param name="parameters">Request parameters</param>
+        /// <param name="requestContext">Request Context</param>
+        /// <returns></returns>
+        public async Task HandleExternalLanguageDeleteRequest(ExternalLanguageDeleteRequestParams parameters, RequestContext<ExternalLanguageDeleteResponseParams> requestContext)
+        {
+            Logger.Write(TraceEventType.Verbose, "HandleExternalLanguageDeleteRequest");
+            try
+            {
+                ConnectionInfo connInfo;
+                ConnectionServiceInstance.TryFindConnection(
+                    parameters.OwnerUri,
+                    out connInfo);
+                ExternalLanguageDeleteResponseParams response = new ExternalLanguageDeleteResponseParams
+                {
+                };
+
+                if (connInfo == null)
+                {
+                    await requestContext.SendError(new Exception(SR.ProfilerConnectionNotFound));
```

Comment:
Wrong error message?

Reply:
And in other ones below too

Reply:
no if the connection not found, that's right generic the error

---
In reply to: [374374133](https://github.com/microsoft/sqltoolsservice/pull/918#discussion_r374374133) [](ancestors = 374374133)